.github/workflows/zip.yml

```diff
@@ -94,6 +94,7 @@ jobs:
            build-head \
            ${{ matrix.linking_type }}
     - uses: actions/upload-artifact@v4
+      if: ${{ always() }}
```

Comment:
How does `always` reconcile with only on failures?

Reply:
My intention is to upload on both success and failures. Currently, the workflow will upload on success implicitly. Using `always()` should cause the upload when the previous building step fails.

Reply:
Makes sense. Thanks